all/gte-10.yml

```diff
@@ -0,0 +1,6 @@
+node_js:
+  - 10
+  - 11
+  - 12
+  - 13
+  - 14
```

Comment:
this could import gte-11, etc, so they're all composed together - is there a reason you didn't go that route?

Reply:
That did not occur to me, TBH - I kept it simple and explicit.

I figure explicit is better than indirect for someone looking at the file as well - no need to go click through and trace things?

Doubt performance is major issue, but fewer imports should be faster too?

Reply:
I doubt it affects performance much; i have a chain of imports to build up almost 300 jobs on all 250+ of my projects and it works fine :-)

as for explicit/implicit, that's a fair point - i'm more thinking of the likelihood of an error in one file being noticed, versus in the chain of imports it'd be very noticeable.

Reply:
Yeah, fair point on error tracking, but that's probably more of an issue when the files are a little more complex than what we have here?

Having errors in these files would have a pretty bad knock on effect, if the ecosystem adopts this, but we can mitigate against that:

1. I do have a generator script; it's pretty trivial, so I didn't bother putting it anywhere else than a branch on my fork. I don't know where to put it just yet, but unit tests on it would be the first life of defense. Alternatively, I'm pondering if I can force renovate to update these files and avoid having to think about where to place the script.
2. There should be an "integration" test as well. One idea I have is to add a `.travis.yml` in here, which just imports all the available files via local import (`node --version` or smth as a test script itself). It shouldn't matter if Travis deduplicates or duplicates the final list (it has the various merge strategies, but I don't recall anything about deduplication) - doing this would at least guarantee the file validation - I think the job just freezes if it doesn't validate? Might need to work around `You can import up to 25 build configuration snippets in total.`.

Reply:
I've been asking travis for item 2 there for a long time :-/ some kind of `travis reify` command would be amazing (we could build it, it's just better to use the canonical implementation). Certainly a viable (but complex) approach would be generating a build on a throwaway branch, downloading the derived config from it, and then cancelling the build (all theroetically doable via travis' API).

You're also totally right about the 25 import limit, which I've told them repeatedly is way too low, so maybe your item 1 is a more reliable approach.

(To be clear, I'm content to consider this a followon and not a blocker)

Reply:
> some kind of travis reify command would be amazing (we could build it, it's just better to use the canonical implementation)

I did build [some of that](https://github.com/pkgjs/detect-node-support/blob/master/lib/travis/imports.js) into `detect-node-support`. It's not canonical, but I think it's close enough (checked against https://github.com/travis-ci/travis-yaml) and the only missing feature, I think, is the `if` support (cause I didn't need it). There is code in there for the merge strategies, but I didn't check how does fetch things, esp. from private repos 🤔 I wonder if it's possibly to write a CLI wrapper or smth around their `travis-yaml` thing 🤔